torch/cuda/_utils.py

```diff
@@ -30,13 +47,35 @@ def _check_cuda(result: int) -> None:
 
 
 def _get_nvrtc_library() -> ctypes.CDLL:
+    if torch.version.hip:
+        return _get_hiprtc_library()
+
     # Since PyTorch already loads NVRTC, we can use the system library
     # which should be compatible with PyTorch's version
     if sys.platform == "win32":
         return ctypes.CDLL("nvrtc64_120_0.dll")
     else:
         return ctypes.CDLL("libnvrtc.so")
 
+def _get_hiprtc_library() -> ctypes.CDLL:
+    # Since PyTorch already loads NVRTC, we can use the system library
+    # which should be compatible with PyTorch's version
+    if sys.platform == "win32":
+        version_str = ''.join(['0', torch.version.hip[0], '0',  torch.version.hip[2]])
+        lib = ctypes.CDLL(f"hiprtc{version_str}.dll")
+    else:
+        lib = ctypes.CDLL("libhiprtc.so")
+
+    # Provide aliases for HIP RTC functions to match NVRTC API
+    lib.nvrtcGetErrorString = lib.hiprtcGetErrorString
+    lib.nvrtcCreateProgram = lib.hiprtcCreateProgram
+    lib.nvrtcDestroyProgram = lib.hiprtcDestroyProgram
+    lib.nvrtcCompileProgram = lib.hiprtcCompileProgram
+    lib.nvrtcGetPTXSize = lib.hiprtcGetCodeSize
+    lib.nvrtcGetPTX = lib.hiprtcGetCode
```

Comment:
Note the usage of `GetCode/GetCodeSize` instead of `GetPTX/GetPTXSize` as HIP doesn't support PTX.